### PR TITLE
trimmed means

### PR DESCRIPTION
**Pull request by hzzhyj**
The trimmed means is the mean of the n-n\alfa deepest curves, given a certain depth measure. This has been implemented in exploratory.stats

### CODE REVIEW

File: skfda/exploratory/stats/_stats.py [hunk 1]
```diff
@@ -91,3 +91,38 @@ def depth_based_median(fdatagrid, depth_method=modified_band_depth):
 
     # The median is the deepest curve
     return fdatagrid[indices_descending_depth[0]]
+
+
+def trimmed_means(fdatagrid,
```
Comment (vnmabus) [inline skfda/exploratory/stats/_stats.py:96]: Maybe should we use same name of function and parameters as numpy (https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.trim_mean.html)?

Comment (hzzhyj) [reply skfda/exploratory/stats/_stats.py]: done

File: skfda/exploratory/stats/_stats.py [hunk 2]
```diff
@@ -91,3 +91,38 @@ def depth_based_median(fdatagrid, depth_method=modified_band_depth):
...
+    """Compute the trimmed means based on a depth measure.
+
+    The trimmed means consists in computing the mean function without a
+    percentage of least deep curves. That is, we first remove the least deep
+    curves and then we compute the mean as usual.
+
+    Args:
+        fdatagrid (FDataGrid): Object containing different samples of a
+            functional variable.
+        proportiontocut (float): indicates the percentage of functions to
```
Comment (vnmabus) [inline skfda/exploratory/stats/_stats.py:108]: There is a problem I have just noticed. In numpy and scipy, `proportiontocut` refers to both sides, while here we only have one side.

File: skfda/exploratory/stats/_stats.py [hunk 3]
```diff
@@ -91,3 +91,38 @@ def depth_based_median(fdatagrid, depth_method=modified_band_depth):
...
+            dataset.
+        depth_method (:ref:`depth measure <depth-measures>`, optional):
+            Method used to order the data. Defaults to :func:`modified
+            band depth <fda.depth_measures.modified_band_depth>`.
+
+    Returns:
+        FDataGrid: object containing the computed trimmed mean.
+
+    """
+    n_samples_to_keep = int((fdatagrid.n_samples -
```
Comment (vnmabus) [inline skfda/exploratory/stats/_stats.py:119]: You can use `scipy.stats.trim1` to simplify the code even more and make its behaviour as similar as possible with scipy.

File: skfda/exploratory/stats/_stats.py (hunk 2 above)
Comment (hzzhyj) [reply skfda/exploratory/stats/_stats.py:108]: The thing is that with functional data it is not possible to cut on the other direction right? we are ranking the data according to their depth measure.

File: skfda/exploratory/stats/_stats.py (hunk 3 above)
Comment (hzzhyj) [reply skfda/exploratory/stats/_stats.py:119]: I tried this but FDataGrid is not compatible with the internal operations of trim1. Maybe just use what is implemented as it is trivial?

Comment (vnmabus) [reply skfda/exploratory/stats/_stats.py:119]: Ok, but make sure that the rounding is the same.

File: skfda/exploratory/stats/_stats.py (hunk 2 above)
Comment (vnmabus) [reply skfda/exploratory/stats/_stats.py:108]: Right. The problem is the possible confusion for a person used to the scipy `trim_mean`. But the only thing that I can think of is to make a note in the documentation explaining it.

File: skfda/exploratory/stats/_stats.py (hunk 3 above)
Comment (hzzhyj) [reply skfda/exploratory/stats/_stats.py]: Done, i am using the exact same operation now to find the index where we  cut.

File: skfda/exploratory/stats/_stats.py (hunk 2 above)
Comment (hzzhyj) [reply skfda/exploratory/stats/_stats.py:108]: Added the note to the documentation. I think that after all we are treating different types of data and the people using this library should be aware. But more documentation never hurts

Comment (vnmabus) [reply skfda/exploratory/stats/_stats.py:113]: No, I think the note misses the point. The important thing to note here is that the scipy function removes twice the `proportiontocut` while we remove it once.

Comment (hzzhyj) [reply skfda/exploratory/stats/_stats.py:113]: That is true, it does not make sense to remove twice the data in our case as it is unidirectional.

**Pull request merged**